lib/improver/ensemble_copula_coupling/ensemble_copula_coupling.py

```diff
@@ -0,0 +1,533 @@
+        calibrated_forecast_variance_data = (
+            calibrated_forecast_variance.data.flatten())
+
+        result = np.zeros((calibrated_forecast_predictor_data.shape[0],
+                           len(percentiles)))
+
+        # Loop over percentiles, and use a normal distribution with the mean
+        # and variance to calculate the values at each percentile.
+        for index, percentile in enumerate(percentiles):
+            percentile_list = np.repeat(
+                percentile, len(calibrated_forecast_predictor_data))
+            result[:, index] = norm.ppf(
+                percentile_list, loc=calibrated_forecast_predictor_data,
+                scale=np.sqrt(calibrated_forecast_variance_data))
+            # If percent point function (PPF) returns NaNs, fill in
+            # mean instead of NaN values. NaN will only be generated if the
+            # variance is zero. Therefore, if the variance is zero, the mean
+            # value is used for all gridpoints with a NaN.
+            if np.any(calibrated_forecast_variance_data == 0):
+                nan_index = np.argwhere(np.isnan(result[:, index]))
+                result[nan_index, index] = (
+                    calibrated_forecast_predictor_data[nan_index])
+            if np.any(np.isnan(result)):
+                msg = ("NaNs are present within the result for the {} "
+                       "percentile. Unable to calculate the percent point "
+                       "function.")
+                raise ValueError(msg)
+
+        result = result.T
+
+        # Reshape forecast_at_percentiles, so the percentiles dimension is
+        # first, and any other dimension coordinates follow.
+        shape_to_reshape_to = list(calibrated_forecast_predictor.shape)
+        if calibrated_forecast_predictor.coord_dims("realization"):
+            realization_coord_position = (
+                calibrated_forecast_predictor.coord_dims("realization"))
+            shape_to_reshape_to.pop(realization_coord_position[0])
+        shape_to_reshape_to = [len(percentiles)] + shape_to_reshape_to
+
+        result = result.reshape(shape_to_reshape_to)
+
+        for template_cube in calibrated_forecast_predictor.slices_over(
+                "realization"):
+            template_cube.remove_coord("realization")
+            break
+        percentile_cube = create_cube_with_percentiles(
+            percentiles, template_cube, result)
+
+        percentile_cube.cell_methods = {}
+        return percentile_cube
+
+    def process(self, calibrated_forecast_predictor_and_variance,
+                raw_forecast):
+        """
+        Generate ensemble percentiles from the mean and variance.
+
+        Parameters
+        ----------
+        calibrated_forecast_predictor_and_variance : Iris CubeList
+            CubeList containing the calibrated forecast predictor and
+            calibrated forecast variance.
+        raw_forecast : Iris Cube or CubeList
```

Comment:
At the moment, this functionality hasn't been added, as it is specific to the generation of percentiles from mean and variance, as required within ensemble calibration. It's not clear that this functionality is required right now, so it might be out of scope for this ticket.